components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Registry/src/Services/DiscovererRegistry.cs

```diff
@@ -221,14 +221,25 @@ public sealed class DiscovererRegistry : IDiscovererRegistry {
                 }
             }
 
-            var queryResult = await _iothub.QueryDeviceTwinsAsync(query, null, pageSize, ct);
-            return new DiscovererListModel {
-                ContinuationToken = queryResult.ContinuationToken,
-                Items = queryResult.Items
+            try {
+                var queryResult = await _iothub.QueryDeviceTwinsAsync(query, null, pageSize, ct);
+                return new DiscovererListModel {
+                    ContinuationToken = queryResult.ContinuationToken,
+                    Items = queryResult.Items
                     .Select(t => t.ToDiscovererRegistration())
                     .Select(s => s.ToServiceModel())
                     .ToList()
-            };
+                };
+            }
+            catch (Exception e) {
```

Comment:
This is not a good fix - let's fix the exception status serialization problem instead.

Reply:
I'm not sure it's a serialization issue. I put breakpoints in ExceptionConverter.cs and in MessagePackSerializer.cs in ExceptionResolver and were never hit.
The error is returned in private async Task<IHttpResponse> SendAsync(IHttpRequest httpRequest, HttpMethod httpMethod, CancellationToken ct) in HttpClient.cs:
![image](https://user-images.githubusercontent.com/7942439/87173902-b2fdf880-c2d6-11ea-83e6-a53db98cd64b.png)

The client.SendAsync return a right response with StatusCode='InternalServerError' and Content empty. 
Content field is what then cause an exception message empty. I was not able to find any serialization issue.

Reply:
![image](https://user-images.githubusercontent.com/7942439/87551834-30e44a00-c6b1-11ea-9ea6-fce83feb164f.png)

the line highlighted in blue is the POST request that causes the error. The answer is empty, and it is what I see in the code

Reply:
![image](https://user-images.githubusercontent.com/7942439/87552205-a51eed80-c6b1-11ea-96bc-cb1d65ab32ad.png)

this is the same POST message with the right response